builtin/providers/aws/resource_aws_db_parameter_group.go

```diff
@@ -161,18 +161,29 @@ func resourceAwsDbParameterGroupUpdate(d *schema.ResourceData, meta interface{})
 		}
 
 		if len(parameters) > 0 {
-			modifyOpts := rds.ModifyDBParameterGroupInput{
-				DBParameterGroupName: aws.String(d.Get("name").(string)),
-				Parameters:           parameters,
-			}
-
-			log.Printf("[DEBUG] Modify DB Parameter Group: %s", modifyOpts)
-			_, err = rdsconn.ModifyDBParameterGroup(&modifyOpts)
-			if err != nil {
-				return fmt.Errorf("Error modifying DB Parameter Group: %s", err)
+			// We can only modify 20 parameters at a time, so walk them until
+			// we've got them all.
+			maxParams := 20
+			for parameters != nil {
+				paramsToModify := make([]*rds.Parameter, 0)
+				if len(parameters) <= maxParams {
+					paramsToModify, parameters = parameters[:], nil
+				} else {
+					paramsToModify, parameters = parameters[:maxParams], parameters[maxParams:]
+				}
+				modifyOpts := rds.ModifyDBParameterGroupInput{
+					DBParameterGroupName: aws.String(d.Get("name").(string)),
+					Parameters:           paramsToModify,
+				}
+
+				log.Printf("[DEBUG] Modify DB Parameter Group: %s", modifyOpts)
+				_, err = rdsconn.ModifyDBParameterGroup(&modifyOpts)
+				if err != nil {
+					return fmt.Errorf("Error modifying DB Parameter Group: %s", err)
```

Comment:
Yeah that actually occurred to me and I couldn't really think of a good solution. I'm totally happy to refactor into a better way if someone has one; I just couldn't figure out a saner way to go.

That written, the read executed about the parameters during plan and apply should reconcile everything. If we completely lost parameter state because we suffered an intermittent connection error or something, we could put them in a partial applied state, but a rerun shouldn't be dangerous or result in a state error.